src-cpp/app/modules/main/module.h

```diff
@@ -26,14 +27,14 @@ class Module : public QObject, virtual public IModuleAccess
     std::unique_ptr<Modules::Main::Wallet::IWalletModuleAccess> m_walletModulePtr;
 
     void connect();
+    void checkIfModuleDidLoad();
```

Comment:
interesting, let me check.

Reply:
btw this would mean blocking our application at the point where we call wait, signals and slots make it properly async and its Qt functionality for async. in order to make it happen here we will anyway have to create a function call back and pass it along with the call and wait for it to be hit.

Reply:
I'd say that currently everything is blocking so it would still be a step forward! but you are right. when calling await, that will block the app and using a callback might works

Another solution is that load function return a sequence of future. That way we only block the app once. not sure if it is legit to block when everything is loading.

Reply:
yes, I think when load things should not be blocked, but once everything is ready in its own time it updates and we know its all loaded. Currently its not all blocked, with signals and slots we achieve the c++'s function callback

Reply:
No reasons to make it `async`.

Reply:
@Khushboo-dev-cpp Qt has synchronisation between threads by `Qt::ConnectionType`. But direct call like `.` or `->` never call it in another thread.